### ISSUE TITLE
Crash upon joining game wheres cheater or somebody with cheating mods.

### ISSUE THREAD

**Issue by nobodySophie** [soumaxetuirk/PD-AntiCheat#4]
### Hi, each time i join someones lobby to try out the punishment system it crashes my game after around 5s from joining.
_I tried removing delayedcallsfix mod cause i thought it could be the reason but it didnt help._

```
Application has crashed: C++ exception
mods/PD AntiCheat/mod/mod.lua:113: attempt to call local 'player_unit' (a userdata value)

SCRIPT STACK

Update() @mods/base/req/core/DelayedCalls.lua:24
func() @mods/base/req/core/DelayedCalls.lua:10
Call() @mods/base/req/core/Hooks.lua:109
update() @mods/base/lua/GameSetup.lua:6
update() lib/setups/networkgamesetup.lua:21
core/lib/setups/coresetup.lua:557
```

**Comment by soumaxetuirk** [comment]
Thank you.
a new checking method implemented on previous update was causing issue for few users.
This is supposed to be fixed now with release 2.3 . Please update your mod and let us know.

**Comment by nobodySophie** [comment]
Works perfectly! 

https://prnt.sc/txc1di - such a wonderful view
